specification/network/resource-manager/Microsoft.Network/preview/2022-06-01-preview/networkVerifier.json

```diff
@@ -0,0 +1,1422 @@
+            "pattern": "^[a-zA-Z0-9]*$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "analysisIntent",
+            "schema": {
+              "$ref": "#/definitions/AnalysisIntent"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "AnalysisIntent_Get": {
+            "$ref": "./examples/AnalysisIntent_Get.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "AnalysisIntent"
+        ],
+        "summary": "Creates the specific Analysis Intent resource",
+        "operationId": "AnalysisIntent_Create",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "in": "path",
+            "name": "networkVerifierName",
+            "description": "The name of the Network Verifier resource",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9]*$"
+          },
+          {
+            "in": "path",
+            "name": "analysisIntentName",
+            "description": "The name of the Analysis Intent resource",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9]*$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "The Analysis Intent resource object to create/update",
+            "schema": {
+              "$ref": "#/definitions/AnalysisIntent"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "analysisIntent was created",
+            "schema": {
+              "$ref": "#/definitions/AnalysisIntent"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
```

Comment:
Define the endpoint for polling the resource by operation id asynchronously. Or, is the final-state-via is supposed be location?

Reply:
Hi @sjanamma, please correct my understanding if I am wrong, but we are returning the full Operation-Location header which the customer can use a GET call on to get the operation status. The Operation-Location header includes the absolute URL with api-version to query. Is this not correct? I was basically following this: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#put-operation-with-additional-long-running-processing (the YOU SHOULD bullets)

Reply:
@arjun-d-patel - Thanks for the clarification. I think we are good here